File: sgps-api-master/src/main/java/br/com/henrique/sgps/domain/ProcessoSeletivo.java
package br.com.henrique.sgps.domain;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class ProcessoSeletivo implements Serializable {

    @Serial
    private static final long serialVersionUID = -871773016390593732L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column
    private String edital;
    @Column
    private String cargo;
    @Column(name = "ano_referencia")
    private Integer anoReferencia;
    @Column(name = "dt_inicio_inscricoes")
    private LocalDateTime dataInicioInscricoes;
    @Column(name = "dt_fim_inscricoes")
    private LocalDateTime dataFimInscricoes;
    @Column(name = "dt_inicio_retificacao")
    private LocalDateTime dataInicioRetificacao;
    @Column(name = "dt_fim_retificacao")
    private LocalDateTime dataFimRetificacao;
    @Column(name = "path_pdf")
    private String pathPDF;

    @Lob
    private byte[] resultado;

    @OneToMany(mappedBy = "processoSeletivo")
    private Set<Inscricao> inscricoes;

}

File: sgps-app/lib/domain/models/processo_seletivo.dart
class ProcessoSeletivo {
  int? id;
  String? edital;
  String? cargo;
  int? anoReferencia;
  String? dataInicioInscricoes;
  String? dataFimInscricoes;
  String? dataInicioRetificacao;
  String? dataFimRetificacao;
  String? pathPdf;

  ProcessoSeletivo(
      {this.id,
      this.edital,
      this.cargo,
      this.anoReferencia,
      this.dataInicioInscricoes,
      this.dataFimInscricoes,
      this.dataInicioRetificacao,
      this.dataFimRetificacao,
      this.pathPdf});

  ProcessoSeletivo.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    edital = json['edital'];
    cargo = json['cargo'];
    anoReferencia = json['anoReferencia'];
    dataInicioInscricoes = json['dataInicioInscricoes'];
    dataFimInscricoes = json['dataFimInscricoes'];
    dataInicioRetificacao = json['dataInicioRetificacao'];
    dataFimRetificacao = json['dataFimRetificacao'];
    pathPdf = json['pathPdf'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['edital'] = edital;
    data['cargo'] = cargo;
    data['anoReferencia'] = anoReferencia;
    data['dataInicioInscricoes'] = dataInicioInscricoes;
    data['dataFimInscricoes'] = dataFimInscricoes;
    data['dataInicioRetificacao'] = dataInicioRetificacao;
    data['dataFimRetificacao'] = dataFimRetificacao;
    data['pathPdf'] = pathPdf;
    return data;
  }
}
